QuantumToolbox.jl/time_evolution/Dicke.qmd

```diff
@@ -0,0 +1,199 @@
+- [`plot_fock_distribution`](https://qutip.org/QuantumToolbox.jl/stable/resources/api#QuantumToolbox.plot_fock_distribution)
+- [`entropy_mutual`](https://qutip.org/QuantumToolbox.jl/stable/resources/api#QuantumToolbox.entropy_mutual)
+
+to explore the Dicke model.
+
+## Introduction
+
+The Dicke model describes a system where $N$ two-level atoms interact with a single quantized electromagnetic mode. The original microscopic form of the Dicke Hamiltonian is given by:
+
+$$
+\hat{H}_D = \omega \hat{a}^\dagger \hat{a} + \sum_{i=1}^{N} \frac{\omega_0}{2} \hat{\sigma}_z^{(i)} + \sum_{i=1}^{N} \frac{g}{\sqrt{N}} (\hat{a} + \hat{a}^\dagger) (\hat{\sigma}_+^{(i)} + \hat{\sigma}_-^{(i)})
+$$
+
+where:
+
+- $\hat{a}$ and $\hat{a}^\dagger$ are the cavity (with frequency $\omega$) annihilation and creation operators, respectively.
+- $\hat{\sigma}_z^{(i)}, \hat{\sigma}_\pm^{(i)}$ are the Pauli matrices for the $i$-th two-level atom, with transition frequency $\omega_0$.
+- $g$ represents the light-matter coupling strength.
+
+This formulation explicitly treats each spin individually. However, when the atoms interact identically with the cavity, we can rewrite the Hamiltonian regarding collective spin operators.
+
+$$
+\hat{J}_z = \sum_{i=1}^{N} \hat{\sigma}_z^{(i)}, \quad \hat{J}_{\pm} = \sum_{i=1}^{N} \hat{\sigma}_{\pm}^{(i)}, 
+$$
+which describe the total spin of the system as a pseudospin of length $j = N/2$. Using these collective operators, the reformulated Dicke Hamiltonian takes the form:
+
+$$
+\hat{H}_D = \omega \hat{a}^\dagger \hat{a} + \omega_0 \hat{J}_z + \frac{g}{\sqrt{N}} (\hat{a} + \hat{a}^\dagger) (\hat{J}_+ + \hat{J}_-)
+$$
+
+This formulation reduces complexity, as it allows us to work on a collective basis instead of the whole individual spin Hilbert space. The superradiant phase transition occurs when the coupling strength $g$ exceeds a critical threshold $g_c = 0.5\sqrt{\omega/\omega_0}$, leading to a macroscopic population of the cavity mode.
+
+
+## Code demonstration
+
+```{julia}
+using QuantumToolbox
+using CairoMakie
+```
+
+```{julia}
+ω = 1
+ω0 = 1
+
+gc = √(ω/ω0)/2
+
+κ = 0.05;
+```
+
+Here, we define some functions for later usage.
+```{julia}
+# M: cavity Hilbert space truncation, N: number of atoms
+Jz(M, N) = (qeye(M) ⊗ jmat(N/2, :z))
+
+a(M, N) = (destroy(M) ⊗ qeye(N+1))
+
+function H(M, N, g)
+    j = N / 2
+    n = 2 * j + 1
+
+    a_ = a(M, N)
+    Jz_ = Jz(M, N)
+    Jp = qeye(M) ⊗ jmat(j, :+)
+    Jm = qeye(M) ⊗ jmat(j, :-);
+
+    H0 = ω * a_' * a_ + ω0 * Jz_
+    H1 = 1/ √N * (a_ + a_') * (Jp + Jm)
+
+    return (H0 + g * H1)
+end;
+```
+
+Take the example of 4 atoms.
+```{julia}
+M0, N0 = 16, 4
+
+H0(g) = H(M0, N0, g)
+
+a0 = a(M0, N0)
+Jz0 = Jz(M0, N0);
+```
+
+```{julia}
+gs = 0.0:0.05:1.0
+ψGs = Vector{QuantumObject{KetQuantumObject}}()
+for _g in gs
+    _H = H0(_g)
+    _, _ψ = eigenstates(_H)
+    push!(ψGs, _ψ[1])
+end
```

Comment:
It is not recommended to write the code like this, as it leads to type instabilities and bad performance. Maybe something like

```suggestion
ψGs = map(gs) do g
    H = H0(g)
     vals, vecs = eigenstates(_H)
     vecs[1]
end
```